packages/grid/_modules_/grid/hooks/features/sorting/useSorting.ts

```diff
@@ -236,6 +236,14 @@ export const useSorting = (apiRef: ApiRef) => {
   const sortApi: SortApi = { getSortModel, setSortModel, onSortModelChange, applySorting };
   useApiMethod(apiRef, sortApi, 'SortApi');
 
+  React.useEffect(() => {
+    // When the rows prop change, we reset the sortedRows state.
+    setGridState((state) => ({
+      ...state,
+      sorting: { ...state.sorting, sortedRows: rowsProp.map((row) => row.id) },
+    }));
+  }, [apiRef, forceUpdate, rowsProp, setGridState]);
```

Comment:
Why do we need the `apiRef` and the `forceUpdate` here as a dependency? You only want to run this if the `rowsProp` changes right?

Reply:
Nicely spotted :)
I forgot them when I was trying other possibilities